mh/pat.py

```diff
@@ -2198,6 +2198,13 @@ def sendAnsLayerCreateHead(self, number, seq):
         TR: Layer creation response
         """
         data = struct.pack(">H", number)
+        if not self.session.is_city_empty(number):
+            self.sendAnsAlert(PatID4.AnsLayerCreateHead, "<LF=8><BODY><CENTER>City already exists.<END>", seq)
+            return
+        if self.session.is_city_reserved(number):
+            self.sendAnsAlert(PatID4.AnsLayerCreateHead, "<LF=8><BODY><CENTER>City reserved.<END>", seq)
+            return
+        self.session.reserve_city(number, True)
```

Comment:
Can't `reserve_city` returns a boolean to show when it's successful? I doubt there is another place where we'd need such check. It will also save a call to the database.

```python
if not self.session.reserve_city(number, True):
    self.sendAnsAlert(PatID4.AnsLayerCreateHead, "<LF=8><BODY><CENTER>City already reserved.<END>", seq)
    return
```

Reply:
Addressed in latest commit